secio/src/codec/mod.rs

```diff
@@ -35,7 +35,7 @@ mod encode;
 /// Type returned by `full_codec`.
 pub type FullCodec<S> = DecoderMiddleware<EncoderMiddleware<length_delimited::Framed<S>>>;
 
-pub type StreamCipher = StreamCipherCore;
+pub type StreamCipher = Box<dyn StreamCipherCore + Send>;
```

Comment:
is this a nightly thing?

Reply:
It was stabilized in Rust 1.27. Not using `dyn` will become deprecated at the end of the year I think.

Reply:
Should we note somewhere that we are expecting you to use latest rust stable (or at least 1.27+)?